src/WaveTrack.cpp

```diff
@@ -1708,9 +1708,10 @@ void WaveTrack::Join(double t0, double t1)
    if( clipsToDelete.size() == 0 )
       return;
 
-   auto t = clipsToDelete[0]->GetSequenceStartTime();
+   auto t = clipsToDelete[0]->GetPlayStartTime();
    //preserve left trim data if any
-   newClip = CreateClip(t, clipsToDelete[0]->GetName());
+   newClip = CreateClip(clipsToDelete[0]->GetSequenceStartTime(),
```

Comment:
A new clip has zero length, `WaveClip::Paste` then makes some checks to decide which data would be discarded, and in that case, the outcome depends on the order in which these cases are handled. So I guess I need to handle another special case in `WaveClip::Paste` so that pasted clip is fully preserved when the target clip is empty. Now if I make a selection strictly inside one clip, and then call join, then the hidden data on the right is lost.

Reply:
> So I guess I need to handle another special case in `WaveClip::Paste` so that pasted clip is fully preserved when the target clip is empty. Now if I make a selection strictly inside one clip, and then call join, then the hidden data on the right is lost.

An alternative fix for the reported bug, would have been to do something similar to void WaveTrack::MergeClips().
That is, not to delete clipsToDelete[0], but to use that as the "newClip", and paste the rest (+silence where necessary). I think this would avoid pasting into an empty clip.

Thanks for approving the  commit!